Application/EdFi.Ods.Api/Security/Authorization/Repositories/RepositoryOperationAuthorizationDecoratorBase.cs

```diff
@@ -318,34 +318,51 @@ string BuildExistenceCheckSql(AuthorizationStrategyFilterResults[] resultsWithPe
                 // Build the existence check SQL
                 StringBuilder sql = new();
 
+                bool queryOptimized=false;
+
                 sql.Append("SELECT CASE WHEN ");
 
                 resultsWithPendingExistenceChecks.ForEach(
                     (x, i, s) =>
                     {
-                        if (i > 0)
+                    if (i > 0)
+                    {
+                        if (x.Operator == FilterOperator.And)
                         {
-                            if (x.Operator == FilterOperator.And)
-                            {
-                                s.Append(" AND ");
-                            }
-                            else
-                            {
-                                s.Append(" OR ");
-                            }
+                            s.Append(" AND ");
                         }
+                        else
+                        {
+                            s.Append(" OR ");
+                        }
+                    }
 
-                        s.Append('(');
-
-                        x.FilterResults.ForEach(
+                    x.FilterResults.ForEach(
+                      (y) =>
+                      {
+                          var viewBasedFilterDefinition = y.FilterDefinition as ViewBasedAuthorizationFilterDefinition
+                              ?? throw new InvalidOperationException(
+                                  "Expected a ViewBasedAuthorizationFilterDefinition instance for performing existence checks.");
+
+                          var viewSqlSupport = viewBasedFilterDefinition.ViewBasedSingleItemAuthorizationQuerySupport;
+
+                          if (x.FilterResults.Any(a => a.FilterDefinition.FilterName == "ClaimEducationOrganizationIdsToStudentUSI" && a.FilterDefinition.FilterName == "ClaimEducationOrganizationIdsToContactUSI"))
```

Comment:
I think this logic belongs outside of a `ForEach` loop as a stand alone statement before you loop through `x.FilterResults` building your query in the statement below.